snakesist/exist_client.py

```diff
@@ -303,24 +376,55 @@ def update_resource(
                 return update replace $node with {updated_node}"""
             )
 
-    def delete_resource(
-        self, abs_resource_id: str, node_id: str = ""
+    def update_node(
+        self, updated_node: str, abs_resource_id: str, node_id: str
     ) -> None:
         """
-        Remove a database resource.
+        Replace a sub-document node with an updated version.
+
+        :param updated_node: The node to replace the old one with.
+        :param abs_resource_id: The absolute resource ID pointing to the document containing the node.
+        :param node_id: The node ID locating the node inside the containing document.
+        """
+        xq = (
+            f"update replace util:node-by-id(util:get-resource-by-absolute-id({abs_resource_id}), '{node_id}')"
+            f"with {updated_node}"
+        )
+        self.query(xq)
+
+    def update_document(
+        self, updated_node: str, path: str
```

Comment:
it's not a node in this case, maybe just call that second argument `data`, it's quiet low-level here anyway.

and, contrary to the supposed insight i stated in an earlier comment, i'm confused by the fact that a document is identified by a path, not an absolute id.

Reply:
i renamed the parameters in the update methods to `data`. 

i hope my reply to that earlier comment sheds a bit of light on the document path issue https://github.com/03b8/snakesist/pull/7#discussion_r329177882